### PR TITLE
chapter_6

### PR DESCRIPTION
**Pull request by kensoloko**

### CODE REVIEW

**Comment by kensoloko** [comment]
ready

File: sample_app/app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX }, uniqueness: { case_sensitive: false }
```
Comment (thangtx88) [inline sample_app/app/models/user.rb]: quá 80 kí tự

**Comment by thangtx88** [comment]
Commented

**Comment by kensoloko** [comment]
ready

File: sample_app/app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+  format: { with: VALID_EMAIL_REGEX }, uniqueness: { case_sensitive: false }
```
Comment (thangtx88) [inline sample_app/app/models/user.rb]: ident

**Comment by thangtx88** [comment]
Commented

**Comment by kensoloko** [comment]
ready

**Comment by thangtx88** [comment]
Merged

**Pull request merged**